trades/finalize_trade_secret.js

```diff
@@ -71,8 +89,68 @@ module.exports = (args, cbk) => {
         return cbk();
       },
 
+      // Get the anchor invoice
+      getAnchorInvoice: ['validate', ({}, cbk) => {
+        // Exit early when there is no anchor invoice (closed trade)
+        if (!args.id) {
+          return cbk(null, {});
+        }
+
+        getInvoice({
+          id: args.id,
+          lnd: args.lnd,
+        },
+        (err, res) => {
+          if (!!err) {
+            return cbk([503, 'UnexpectedErrorGettingAnchorInvoice', err]);
+          }
+          const {request} = res;
+
+          return cbk(null, {request});
+        },
+        cbk);
+      }],
+
+      // Parse anchor invoice
+      decodeAnchorInvoice: ['getAnchorInvoice', ({getAnchorInvoice}, cbk) => {
+        // Exit early when there is no anchor invoice
+        if (!getAnchorInvoice.request) {
+          return cbk(null, {});
+        }
+
+        try {
+          const {request} = getAnchorInvoice;
+          const requestDetails = parsePaymentRequest({request});
```

Comment:
Why does the payreq need to be parsed, if the invoice is local the description should be returned from the get invoice call?